The lock variable in ApiListener::Start is missing its name

```In file included from lib/remote/remote_unity.cpp:5:
../lib/remote/apilistener.cpp:223:28: warning: parentheses were disambiguated as redundant parentheses around declaration of variable named 'm_LogLock' [-Wvexing-parse]
                boost::mutex::scoped_lock(m_LogLock);
                                         ^~~~~~~~~~~
../lib/remote/apilistener.cpp:223:28: note: add a variable name to declare a 'boost::mutex::scoped_lock' (aka 'unique_lock<boost::mutex>') initialized with 'm_LogLock'
                boost::mutex::scoped_lock(m_LogLock);
                                         ^
                                          varname
../lib/remote/apilistener.cpp:223:3: note: add enclosing parentheses to perform a function-style cast
                boost::mutex::scoped_lock(m_LogLock);
                ^
                (                                   )
../lib/remote/apilistener.cpp:223:28: note: remove parentheses to silence this warning
                boost::mutex::scoped_lock(m_LogLock);
                                         ^         ~

1 warning generated.```

--- a/lib/remote/apilistener.cpp
+++ b/lib/remote/apilistener.cpp
@@ -220,7 +220,7 @@ void ApiListener::Start(bool runtimeCreated)
 	ObjectImpl<ApiListener>::Start(runtimeCreated);
 
 	{
-		boost::mutex::scoped_lock(m_LogLock);
+		boost::mutex::scoped_lock lock(m_LogLock);
 		RotateLogFile();
 		OpenLogFile();
 	}

